src/Entities/Entity.cpp

```diff
@@ -19,15 +19,21 @@
 
 
 
-UInt32 cEntity::m_EntityCount = 0;
-cCriticalSection cEntity::m_CSCount;
+static UInt32 GetNextUniqueID(void)
+{
+	static std::atomic<UInt32> counter(1);
+	return counter.fetch_add(1);
+}
 
 
 
 
 
+////////////////////////////////////////////////////////////////////////////////
+// cEntity:
+
 cEntity::cEntity(eEntityType a_EntityType, double a_X, double a_Y, double a_Z, double a_Width, double a_Height):
-	m_UniqueID(INVALID_ID),  // Proper ID will be assigned later in the constructor code
+	m_UniqueID(GetNextUniqueID()),
```

Comment:
Because the `m_EntityCount` would still be a de-facto global object requiring a global constructor.